lib/tests/misc.nix

```
@@ -18,6 +18,26 @@ runTests {
     expected = 2;
   };
 
+  testPipe = {
+    expr = pipe 2 [
+      (x: x + 2) # 2 + 2 = 4
+      (x: x * 2) # 4 * 2 = 8
+    ];
+    expected = 8;
+  };
+
+  testPipeStrings = {
+    expr = pipe [ 3 4 ] [
+      (map toString)
+      (map (s: s + "\n"))
+      concatStrings
+    ];
+    expected = ''
+      3
+      4
+    '';
+  };
+
```

Maybe you add a test that an empty pipe works as expected:

```suggestion

  testEmptyPipe = {
    expr = pipe 2 [];
    expected = 2;
  };
```

---

Thanks, I added the empty pipe test as you suggested.